pkgs/by-name/gn/gnunet-messenger-cli/package.nix

```diff
@@ -0,0 +1,67 @@
+{
+  lib,
+  stdenv,
+  fetchgit,
+  writeScript,
+  meson,
+  ninja,
+  pkg-config,
+  gnunet,
+  libsodium,
+  libgcrypt,
+  libgnunetchat,
+  ncurses,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "gnunet-messenger-cli";
+  version = "0.3.1";
+
+  src = fetchgit {
+    url = "https://git.gnunet.org/messenger-cli.git";
+    tag = "v${finalAttrs.version}";
+    hash = "sha256-8Iby3IZXEZJ1dqVV62xDzXx/qq7JKhVtn6ZLb697ZSw=";
+  };
+
+  INSTALL_DIR = (placeholder "out") + "/";
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+  ];
+
+  buildInputs = [
+    gnunet
+    libgcrypt
+    libgnunetchat
+    libsodium
+    ncurses
+  ];
+
+  passthru.updateScript = writeScript "update-gnunet-messenger-cli" ''
+    #!/usr/bin/env nix-shell
+    #!nix-shell -i bash -p common-updater-scripts curl
+
+    set -euo pipefail
+
+    # Hardcoded latest version (update this when a new tag is released)
+    latest_version="0.3.1"
+
+    # Use the update helper
+    update-source-version gnunet-messenger-cli "$latest_version"
+  '';
```

Comment:
hardcoding the version completely defeats the point of adding an update script fwiw.